tracer/src/Datadog.Trace/ClrProfiler/ServerlessInstrumentation/AWS/LambdaCommon.cs

```diff
@@ -243,5 +257,68 @@ private static void WriteRequestHeaders(WebRequest request, IDictionary<string,
             var proxyInstance = obj.DuckAs<ILambdaContext>();
             return proxyInstance?.ClientContext?.Custom;
         }
+
+        private class ConverterContractResolver : DefaultContractResolver
+        {
+            protected override JsonContract CreateContract(Type objectType)
+            {
+                // Create a contract with custom settings
+                var contract = base.CreateContract(objectType);
+
+                if (objectType == typeof(MemoryStream))
+                {
+                    contract.Converter = MemoryStreamConverter;
+                }
+                else if (objectType == typeof(DateTime))
+                {
+                    contract.Converter = DateTimeConverter;
+                }
+
+                return contract;
+            }
+        }
+
+        private class MemoryStreamJsonConverter : JsonConverter<MemoryStream>
+        {
+            public override void WriteJson(JsonWriter writer, MemoryStream value, JsonSerializer serializer)
+            {
+                if (value == null)
+                {
+                    writer.WriteNull();
+                    return;
+                }
+
+                try
+                {
+                    var base64String = Convert.ToBase64String(value.ToArray());
+                    writer.WriteValue(base64String);
+                }
+                catch (Exception ex)
+                {
+                    Serverless.Debug($"Failed to serialize MemoryStream with the following error: {ex}");
+                    writer.WriteNull();
+                }
+            }
+
+            public override MemoryStream ReadJson(JsonReader reader, Type objectType, MemoryStream existingValue, bool hasExistingValue, JsonSerializer serializer)
+            {
+                throw new NotImplementedException();
+            }
+        }
+
+        private class DateTimeJsonConverter : JsonConverter<DateTime>
+        {
+            public override void WriteJson(JsonWriter writer, DateTime value, JsonSerializer serializer)
+            {
+                var elapsedTime = value - DateTime.UnixEpoch;
+                var timestamp = elapsedTime.TotalSeconds;
```

Comment:
Are we sure this is seconds and not milliseconds?

Reply:
Seconds method will give you milliseconds too, the extension expects the value as "16373.728" where the last part is the milliseconds.

Reply:
Adding a comment about it so it's clear for others.

Reply:
Thanks!